src/test/resources/org/jenkinsci/plugins/scriptsecurity/casc/ScriptApprovalConfiguratorTest.yml

```diff
@@ -0,0 +1,5 @@
+scriptApproval:
+  approvedSignatures:
+    - method java.net.URI getHost
```

Comment:
I do not think this is a good idea. We should not be encouraging administrators to create custom preapprovals. Genuinely safe methods can be added to the `generic-whitelist` (or similar) in this plugin and a new release cut. If it is not obvious whether a method is safe, do not whitelist it. Simplify your scripts to do processing in external processes.

Reply:
do you mean the feature to let administrator add a signature to the whitelist _from the web UI_ is a bad idea ?

Reply:
@jglick, @ndeloof - I understand that having custom approved signatures should not be encouraged, however I think that since this feature exists, it will be better for Administrators to be able to control this configuration within a source code repository and not using the UI. I think that any feature that can be configured through the UI, should have an ability to be configured through source code.
Best,
Ohad

Reply:
Speaking as one who has been deploying jenkins in a test env with puppet for modifications, then pushing those changes into production for 4 years this is not a minor concern.  I have something like 30 signatures to approve -- which means having to run test pipelines after a production push to see if any new signatures are needed and the pain and misery have having to run a pipeline script + approve a signature in a loop 30 times for ever new sandbox env (typically several times per week and sometimes per day).